apps/gnome-mobile/src/components/ui/DraggableGnome.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useRef, useState } from "react";
+import { Alert, Image, PanResponder, View } from "react-native";
+
+const DraggableGnome = () => {
+  const trackWidth = 344;
+  const trackHeight = 70;
+  const gnomeWidth = 100;
+  const gnomeHeight = 63;
+  const startX = -trackWidth / 2 + gnomeWidth / 2;
+  const endX = startX + trackWidth - gnomeWidth + 20;
+
+  const [position, setPosition] = useState({ x: startX, y: 0 });
+
+  const pan = useRef(
+    PanResponder.create({
+      onStartShouldSetPanResponder: () => true,
+      onPanResponderMove: (_, gestureState) => {
+        const newX = Math.max(startX, Math.min(startX + gestureState.dx, endX));
+        setPosition({ x: newX, y: 0 });
+      },
+      onPanResponderRelease: (_, gestureState) => {
+        if (gestureState.dx >= endX - startX - 20) {
+          Alert.alert("Unlocked!");
```

Comment:
tą akcję trzeba jakoś przekazać z zewnątrz do tego komponentu, jak go gdzieś chcemy użyć to fajnie by było móc tam podpiąć jakiegoś callbacka

Reply:
coś jak onClick w buttonach